account_invoice_mgmt_dms/controllers/main.py

```diff
@@ -0,0 +1,26 @@
+from odoo import http, _
+from odoo.http import request
+import base64
+import io
+import zipfile
+
+
+class Binary(http.Controller):
+
+    @http.route('/web/binary/download_purchase_invoice', type='http', auth="public")
```

Comment:
If possible, change URL definition (referring addon name instead of `/web/binary`), and this rout shouldn't be public accessed.